fix(hardware): tolerate move timeout not scheduled

sfoster1 opened the pull request:
If the asyncio coroutine that runs the move ack timeout doesn't get scheduled, we can end up falsely claiming the move timed out. Double check after we get scheduled because of a timeout error that in fact the required acks haven't come in.

sfoster1 committed:
fix(hardware): tolerate move timeout not scheduled

If the asyncio coroutine that runs the move ack timeout doesn't get
scheduled, we can end up falsely claiming the move timed out. Double
check after we get scheduled because of a timeout error that in fact the
required acks haven't come in.

mjhuff reviewed (approved):
Makes sense, thanks!

sfoster1 committed:
merge edge

sfoster1 committed:
format

sfoster1 committed:
toss in one last chance to get the message

sfoster1 committed:
fix(hardware): refactor the runner timeout again

By splitting the timeout on the move group runs into two segments. When
we timeout, we aren't usually actually timing out because the ack took
too long; either it gets or it doesn't. it's just that sometimes we
don't notice that it got there. So what we expect is the first timeout
of minimum 3s will pass unless we don't get scheduled or unless the
message got lost; if it fails, then we want to wait again in case we
didn't get scheduled. And then if the second timeout fails, the message
definitely got lost.

sfoster1 committed:
fixup

sfoster1 committed:
some slightly nicer logging

sfoster1 committed:
nope
